### ISSUE TITLE
Searching in collection views does not search track titles

### ISSUE THREAD

**Issue by jamaggs** [quodlibet/quodlibet#4540]
Steps to reproduce
------------------
When in cover grid and album list, the search box does not seem to search the title of the individual tracks. So, if I search for a specific song it will show zero albums matching. By contrast, if I search for an artist the behavior is that it will search for the artist for the specific tracks and return the album(s) in which the relevant tracks are contained.

If I search "title=<SOME TITLE>" it does search by title and behave as expected, but the documentation suggests that it should not be necessary to use this syntax:

"Pretty much every view in Quod Libet contains a search entry where you can enter search terms and save them. Quod Libet will search in artist, album, title, version, and any other visible columns for what you enter."

Expected Output
---------------
When searching for a specific track, the cover grid and album list should show the albums in which these tracks are contained (searching by title).

Actual Output
-------------
Searching by title returns zero albums.

Test System
-----------

### Which version of Quod Libet?
4.4.0

### Which operating system 
Mint 21.3

### If it's audio-related, what back-end?

Additional Information
----------------------
Many thanks for a great piece of software - it's really very good!

**Comment by declension** [comment]
Understand this is confusing perhaps, but if I'm understanding your issue, this is as designed. A bit of explanation:

The album / collection browsers have search interfaces to _their items_ - i.e. _albums_ here. So that means you are searching _album data_ for that (free) text e.g. `the long and winding road`, and no albums have **exposed** (to the browser) information matching that. I say exposed to outline that QL has a paradigm of searching _visible columns_ (in the track browser), or more generally _exposed_ data around that view. And track titles are not included in that view of albums

However, when you use a structured query, e.g. `title=the long and winding road`, that gets more interesting. QL explicitly searches each item (~=album), and due to the clever-but-confusing aggregation applied, you get "the expected" behaviour generally (but see [docs on Internal Album Tags](https://quodlibet.readthedocs.io/en/latest/guide/tags/internal_tags.html#id2)), which in non-exceptional cases like `title`, [gets the union of all values](https://github.com/quodlibet/quodlibet/blob/d09590aacae1740742f6539f7764eb4737c16285/quodlibet/util/collection.py#L285-L293)

So then, you *will* match an album with that track title in, e.g. _The Beatles - Let it Be_ in this example.

Now if we were to change this (and only presumably for the special case of `title` - `genre` or `comment` would probably be madness), it'd a pretty bad behavioural change for all users in many cases (suddenly it will be very hard to find some albums with short / common titles, because _any album with any tracks with those words would match), but I guess a toggle button could mitigate / fix this...

**Comment by declension** [comment]
> Also, it seems to search the track artists (not just the album artist) when I’m album view which seems a little inconsistent.

Yes - artist is a value exposed in the album items, and will be the union of all artists across the tracks. But if you name a track in your album, that album doesn't "inherit" this name in any way (but it does inherit the artists that worked on that track)

This still seems like the right behaviour but it should be better documented, and the enhancement as above might be nice too.
